hrjobcontract/js/src/job-contract/services/utils.service.js

```diff
@@ -72,6 +73,40 @@ define([
         return deffered.promise;
       },
 
+      getOptionValues: function (groupNames) {
+        var deferred = $q.defer();
+        var multiple = _.isArray(groupNames);
+
+        CRM.api3('OptionValue', 'get', {
+          'sequential': 1,
+          'is_active': 1,
+          'option_group_id.name': { 'IN': multiple ? groupNames : [groupNames] },
+          'return': [ 'id', 'label', 'weight', 'value', 'is_active', 'option_group_id', 'option_group_id.name' ],
+          'options': {
+            'limit': 1000,
+            'sort': 'id'
```

Comment:
is the sort really necessary?

Reply:
Sorting is not necessary. This will be catered for by UI-Select. Replacing the whole implementation with API.resource as suggested.